File: src/euler_007.hs
isqrt :: Int -> Int
isqrt = floor . sqrt . fromIntegral

factors :: Int -> [Int]
factors n = [x | x <- [2..isqrt n], n `mod` x == 0]

isPrime :: (Int) -> Bool
isPrime n = factors n == []

answer :: Int -> Int
answer n = last $ take n [x | x <-[2..], isPrime x]
main = putStrLn $ show $ answer 10001

File: src/euler_007.jl
is_prime(n) = map(x -> n % x != 0, 2:isqrt(n)) |> all

function answer(n)
    filter(is_prime, 3:2:200_000) |> x -> getindex(x, n - 1)
end

answer(10_001) |> print
